ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,448 @@
+                using (var archiveStream = StreamProvider.ForExtension(".gz").Open(archive.FullName).First())
+                {
+                    Convert(archiveStream);
+                }
+
+                Log.Trace($"PsychSignalDataConverter.ConvertBackfill(): Finished converting historical data for file: {archive.FullName}");
+            }
+
+            Dispose();
+        }
+
+        /// <summary>
+        /// Iterate over the data, processing each data point before finally zipping the directories
+        /// </summary>
+        public void ConvertDirectory()
+        {
+            if (_disposedValue)
+            {
+                throw new ObjectDisposedException("PsychSignalDataConverter has already been disposed");
+            }
+
+            // Filter for raw data with file names formatted as "yyyyMMdd_HH.*"
+            // `GetDirectory()` doesn't guarantee file order, so we must order it manually ourselves
+            var files = _rawSourceDirectory.GetFiles("*.csv", SearchOption.TopDirectoryOnly)
+                .Where(
+                    x =>
+                    {
+                        DateTime fileDate;
+                        return DateTime.TryParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal, out fileDate);
+                    }
+                )
+                .OrderBy(x => DateTime.ParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal))
+                .ToList();
+
+            var i = 0;
+            var fileCount = files.Count();
+
+            foreach (var rawFile in files)
+            {
+                i++;
+                Log.Trace($"PsychSignalDataConverter.ConvertFrom(): Reading file {rawFile.Name} (file {i}/{fileCount})");
+                Log.Trace($"PsychSignalDataConverter.ConvertDirectory(): Begin converting {rawFile.Name}");
+
+                using (var stream = rawFile.OpenRead())
+                {
+                    Convert(stream);
+                }
+
+                Log.Trace("PsychSignalDataConverter.ConvertDirectory(): Finished converting {rawFile.Name}");
+            }
+
+            Dispose();
+        }
+
+        /// <summary>
+        /// Utility method to compresses the data contained with the psychsignal alternative data folder
+        /// to a structure similar to equity minute files (e.g. /[symbol]/20010101.zip#20010101.csv)
+        /// </summary>
+        private void CompressData()
+        {
+            if (_disposedValue)
+            {
+                throw new ObjectDisposedException("PsychSignalDataConverter has already been disposed");
+            }
+
+            Log.Trace("PsychSignalDataConverter.CompressData(): Begin compressing PsychSignal data");
+            var timer = Stopwatch.StartNew();
```

Comment:
`Stopwatch` is great for things that operate on sub-second intervals -- typically for anything that's interacting with I/O, using a `var start = DateTime.UtcNow` is a little more lightweight.